spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelBindingUtils.java

```diff
@@ -0,0 +1,67 @@
+package org.springframework.cloud.stream.binding;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.cloud.stream.binder.DirectHandler;
+import org.springframework.integration.config.ConsumerEndpointFactoryBean;
+import org.springframework.integration.scheduling.PollerMetadata;
+import org.springframework.messaging.PollableChannel;
+import org.springframework.messaging.SubscribableChannel;
+import org.springframework.scheduling.support.PeriodicTrigger;
+
+/**
+ * An utility class that provides helper methods for channel binding.
+ * This class needs to be instantiated as a bean so that the configuration properties are available runtime.
+ *
+ * @author Marius Bogoevici
+ * @author David Syer
+ * @author Ilayaperumal Gopinathan
+ */
+public class ChannelBindingUtils implements BeanFactoryAware {
```

Comment:
in Spring projects, typically a "Utils" class provides only static methods, rather than being a bean in the application context

at a minimum, we should rename this, but it's worth considering whether this functionality cannot be handled in a different way